### PR TITLE
 feat(FEC-8662): add UI config token evaluation for share

### PR DESCRIPTION
**Pull request by OrenMe**
### Description of the Changes

* add UI config token evaluation - used for share mainly now
* fix tests and added new ones for token evaluation

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/kaltura-player.js [hunk 1]
```diff
@@ -510,6 +512,29 @@ class KalturaPlayer extends FakeEventTarget {
...
   }
+
+  /**
+   * set the share config
+   * @param {string} entryId - the entry id
+   * @returns {void}
+   * @private
+   */
+  _maybeSetEmbedConfig(entryId: string): void {
+    const serviceUrl = this._localPlayer.config.provider.env.serviceUrl.replace('api_v3', '');
```
[the remark this answers is not in the document]
Comment (OrenMe) [reply src/kaltura-player.js]: done

File: src/kaltura-player.js [hunk 2]
```diff
@@ -510,6 +512,29 @@ class KalturaPlayer extends FakeEventTarget {
...
+    const share = this._localPlayer.config.ui.components.share;
```
[the remark this answers is not in the document]
Comment (OrenMe) [reply src/kaltura-player.js]: done

File: src/kaltura-player.js [hunk 3]
```diff
@@ -510,6 +512,29 @@ class KalturaPlayer extends FakeEventTarget {
...
+    let shareUrl = share.shareUrl;
+    if (!shareUrl) {
+      shareUrl = `${serviceUrl}/index.php/extwidget/preview/partner_id/{{partnerId}}/uiconf_id/{{uiConfId}}/entry_id/${entryId}/embed/dynamic`;
```
[the remark this answers is not in the document]
Comment (OrenMe) [reply src/kaltura-player.js]: the service url is passed in the config so it will take the configured one

File: src/kaltura-player.js [hunk 4]
```diff
@@ -510,6 +516,33 @@ class KalturaPlayer extends FakeEventTarget {
...
+    if (ui && ui.components && ui.components.share) {
+      const share = this.config.ui.components.share;
+      let shareUrl = share.shareUrl;
+      let embedUrl = share.embedUrl;
+      const shareConfig = {
+        ui: {
+          components: {
+            share: {
+              ...(shareUrl && shareUrl),
+              ...(embedUrl && embedUrl)
```
Comment (yairans) [inline src/kaltura-player.js]: can you explain the purpose and the duplication?

File: src/common/plugins/plugins-config.js [hunk 5]
```diff
@@ -88,41 +92,82 @@ const getModel = (options: KPOptionsObject): Object => {
...
-          }
+/**
+ * @param {UIOptionsObject} options - UI options
+ * @param {KPOptionsObject} config - player config
+ * @private
+ * @return {void}
+ */
+function evaluateUIConfig(options: UIOptionsObject, config: KPOptionsObject): void {
+  if (options) {
+    const defaultUiConfig = {
```
Comment (yairans) [inline src/common/plugins/plugins-config.js]: consider move to store (like plugins)

File: src/kaltura-player.js (hunk 4 above)
Comment (OrenMe) [reply src/kaltura-player.js]: it's a way to define conditional if to add member to object.
It's the equivalent of writing:
```javascript
if (shareUrl) {
  share.shareUrl = shareUrl;
}
if (embedUrl) {
  share.embedUrl = embedUrl;
}
```

File: src/common/plugins/plugins-config.js (hunk 5 above)
Comment (OrenMe) [reply src/common/plugins/plugins-config.js]: I agree, but we currently have a bug with current implementation, which is shared across player instances so didn't want to recreate it here.
Once we move to new token evaluation we can create one big happy store and fix the static dependency.

File: src/kaltura-player.js (hunk 4 above)
Comment (yairans) [reply src/kaltura-player.js]: for me `...{x && x}` does something else. bad thing:
```
x = 'Oren';
{...(x && x)} // ['O','r','e','n']
```

File: src/setup.js [hunk 6]
```diff
@@ -27,7 +27,9 @@ function setup(options: PartialKPOptionsObject | LegacyPartialKPOptionsObject):
   const defaultOptions = getDefaultOptions(options);
   setLogLevel(defaultOptions);
   printSetupMessages();
-  evaluatePluginsConfig(defaultOptions);
+  if (defaultOptions.plugins) {
```
Comment (yairans) [inline src/setup.js]: not needed. we protect it internally.

File: src/kaltura-player.js [hunk 7]
```diff
@@ -510,6 +516,33 @@ class KalturaPlayer extends FakeEventTarget {
...
+
+  /**
+   * set the share config
+   * @returns {void}
+   * @private
+   */
+  _maybeSetEmbedConfig(): void {
+    const ui = this.config.ui;
+    if (ui && ui.components && ui.components.share) {
+      const share = this.config.ui.components.share;
```
Comment (yairans) [inline src/kaltura-player.js]:
```suggestion
      const share = ui.components.share;
```

File: src/kaltura-player.js [hunk 8]
```diff
@@ -510,6 +516,33 @@ class KalturaPlayer extends FakeEventTarget {
   get Error(): typeof Error {
     return this._localPlayer.Error;
   }
+
+  /**
+   * set the share config
+   * @returns {void}
+   * @private
+   */
+  _maybeSetEmbedConfig(): void {
```
Comment (yairans) [inline src/kaltura-player.js]: Could you explain what you do in this method?
from my understanding, you take values from share config and merge them back to the same room.

**Pull request merged**